core/app/c/[communitySlug]/stages/[stageId]/page.tsx

```diff
@@ -67,7 +85,15 @@ export default async function Page(props: {
 		{ type: MembershipType.community, communityId: community.id },
 		user.id
 	);
-	const [stage, showEditButton] = await Promise.all([stagePromise, capabilityPromise]);
+	const [{ stage, canViewStage }, showEditButton] = await Promise.all([
+		stagePromise,
+		capabilityPromise,
+	]);
+
+	if (!canViewStage) {
+		// TODO: replace with new nextjs auth interrupts
```

Comment:
Could you explain this TODO a little? Not sure what new NextJS auth interrupts are!

Reply:
yeah sorry, it's this: https://nextjs.org/docs/app/api-reference/config/next-config-js/authInterrupts

next introduced the ability to actually "throw" `401`s and `403` using `unauthorized()` and `forbidden()`, a la `notFound()`. i think we should use those instead of this unauthorized page in the future. i started doing that, but thought it would be too much clutter for this pr, so i put that todo there to remind me